### ISSUE TITLE
Variable values not present in the result message of the validation report

### ISSUE THREAD

**Issue by m0squito** [TopQuadrant/shacl#30]
I have developed several SPARQL-based constraints with success. For each of them, I provide a custom `sh:message` with variables (e.g. `{?var1}`) that are replaced with bindings from the SPARQL query, when the validation report is produced.

Today, I developed a first query that used an aggregate function and the variables in the message are not replaced with values. I looked at the [spec](https://www.w3.org/TR/shacl/#sparql-constraints-variables), but it does not mention anything regarding aggregate functions. Perhaps, I did something wrong here -- any ideas?

Here's the shape:
```Turtle
cfgs:TotalWeightShape
        a sh:SPARQLConstraint ;
	sh:message "The total weight of {$this} ({?value}) exceeds the maximum of 20." ;
	sh:prefixes cfgs: ;
	sh:select """
```
```SPARQL 
                SELECT $this (sum(?weight) AS ?value)
		WHERE {			
			$this cfg:hasPart/cfg:hasWeight ?weight .
		}
		GROUP BY $this
		HAVING (?value > 20)
	""" .
```
Here's the report produced by the API:
```Turtle
  sh:result    [ a                             sh:ValidationResult ;
                 sh:focusNode                  cfg:Config_2 ;
                 sh:resultMessage              "The total weight of {$this} {?value} exceeds the maximum of 20." ;
                 sh:resultSeverity             sh:Violation ;
                 sh:sourceConstraint           cfgs:TotalWeightShape ;
                 sh:sourceConstraintComponent  sh:SPARQLConstraintComponent ;
                 sh:sourceShape                cfgs:TotalWeightShape ;
                 sh:value                      54
               ]
```
Note that `sh:value` is properly bound to `54`.

**Comment by HolgerKnublauch** [comment]
This works for me (the numbers get inserted into the message), using the latest snapshot (and called from TopBraid although I believe the behavior should be the same from the command line). Could you clarify how you are invoking this and what version you have tried - did you try rebuilding from source?

**Comment by m0squito** [comment]
I run it in Java, using the 1.0.0 release:

```Java
	public void run(File dataFile, File shapeFile) throws FileNotFoundException {
		OntModel dataModel = ModelFactory.createOntologyModel(OntModelSpec.OWL_MEM);
		OntModel shapeModel = ModelFactory.createOntologyModel(OntModelSpec.OWL_MEM);

		dataModel.read(new FileInputStream(dataFile), "", FileUtils.langXML);
		shapeModel.read(new FileInputStream(shapeFile), "urn:x-base", FileUtils.langTurtle);

		Resource report = ValidationUtil.validateModel(dataModel, shapeModel, true);
		report.getModel().write(System.out, FileUtils.langTurtle);
	}
```

I will create a minimal, complete, and verifiable example tomorrow.

**Comment by m0squito** [comment]
Holger, I figured out what was causing the problem. I had two individuals with the same name, one representing a node shape, and the other representing a SPARQL constraint, tied to the first one. The validator worked well, actually, but it didn't bind the values to variables in the report message. Once I created unique names for both individuals, it worked as expected.

I created a [small example](https://gist.github.com/m0squito/aba366c80ad788a2ffed61134039bbdf) to demonstrate the problem.
